### PR TITLE
Add `prefixSearch` and `facetSearch` index settings

### LINKED ISSUE #694: [v1.12.0] New settings: `facetSearch` & `prefixSearch`
Related to https://github.com/meilisearch/integration-guides/issues/307

### Explanation of the feature

- New index settings: `facetSearch`
- New index settings: `prefixSearch`

Usage: https://meilisearch.notion.site/API-usage-Settings-to-opt-out-indexing-features-fff4b06b651f8108ade3f858aeb16b14

### TODO

- [ ] Ensure the methods `getSettings()`, and `updateSettings` can accept the new settings fields: `facetSearch` & `prefixSearch`
- [ ] Add the new methods: `getFacetSearch()`, `updateFacetSearch()` and `resetFacetSearch()` calling the appropriate Meilisearch routes
- [ ] Add the new methods: `getPrefixSearch()`, `updatePrefixSearch()` and `resetPrefixSearch()` calling the appropriate Meilisearch routes
- [ ] Add tests

⚠️ Make PRs pointing to `bump-meilisearch-v1.12.0` and NOT `main`. Please do 1 PR for all of these changes, and not several.

### PR DESCRIPTION
**Pull request by Strift**
# Pull Request

## Related issue
Fixes #694 

## What does this PR do?
- Add tests to make sure `getSettings()`, and `updateSettings` allow the new settings
- Add the `getFacetSearch()`, `updateFacetSearch()`, and `resetFacetSearch()` methods
- Add the `getPrefixSearch()`, `updatePrefixSearch()`, and `resetPrefixSearch()` methods
- Add tests

### CODE REVIEW

**Comment by Strift** [comment]
Hey @norkunas 👋  I went ahead I tagged you as a reviewer since you helped me with the previous PRs. I hope you can assist me with reviewing this one too. 🙏

File: src/Endpoints/Delegates/HandlesSettings.php [hunk 1]
```diff
@@ -421,4 +421,60 @@ public function resetEmbedders(): array
     {
         return $this->http->delete(self::PATH.'/'.$this->uid.'/settings/embedders');
     }
+
+    // Settings - Facet Search
+
+    /**
+     * @since Meilisearch v1.12.0
```
Comment (norkunas) [inline src/Endpoints/Delegates/HandlesSettings.php:428]: do we really need this annotation (everywhere)?

**Review by norkunas — APPROVED** [review]

File: src/Endpoints/Delegates/HandlesSettings.php (hunk 1 above)
Comment (Strift) [reply src/Endpoints/Delegates/HandlesSettings.php:428]: I wouldn't say we need it to the point where we should update all the code base. But if it shows up in the generated documentation, then it's useful information for the reader. What do you think?

Comment (norkunas) [reply src/Endpoints/Delegates/HandlesSettings.php:428]: Ok

**Pull request merged**